sql/driver/wire.go

```diff
@@ -80,14 +100,17 @@ func (d Datum) Value() (driver.Value, error) {
 		val = t.BytesVal
 	case *Datum_StringVal:
 		val = t.StringVal
+	case *Datum_DateVal:
+		val = Date{t.DateVal.GoTime()}
```

Comment:
The inconsistency I was referring to was that if someone does `select now()::date, now()`, they may get something like ('2015-09-21', '2015-09-20 20:00:00') depending on their timezone. It's going to be confusing that timestamps display in local time but all other date math is done in UTC. I think that returning everything in UTC is a better short-term solution until we can make the server-side date math aware of the user's desired timezone.